contracts/BlendSwap.sol

```diff
@@ -38,80 +39,91 @@ contract BlendSwap {
+            secretHash: secretHash,
+            fee: fee
         });
 
-
-        if (secretHash == 0x00) {
-            status[lockId] = Status.INITIALIZED;
+        if (confirmed) {
+            status[secretHash] = Status.CONFIRMED;
         } else {
-            status[lockId] = Status.HASH_REVEALED;
-            hashlocks[lockId] = secretHash;
+            status[secretHash] = Status.INITIALIZED;
         }
 
-        blend.transferFrom(msg.sender, address(this), amount);
+        blend.transferFrom(msg.sender, address(this), amount + fee);
     }
 
-    function revealSecretHash(bytes32 lockId, bytes32 secretHash) public {
+    function confirmSwap(bytes32 secretHash) public {
         require(
-            status[lockId] == Status.INITIALIZED,
+            status[secretHash] == Status.INITIALIZED,
             "Wrong status"
         );
+
         require(
-            msg.sender == swaps[lockId].from,
+            msg.sender == swaps[secretHash].from,
             "Sender is not the initiator"
         );
 
-        status[lockId] = Status.HASH_REVEALED;
-        hashlocks[lockId] = secretHash;
+        status[secretHash] = Status.CONFIRMED;
     }
 
-    function redeem(bytes32 lockId, bytes32 secret) public {
+    function redeem(bytes32 secret) public {
+        bytes32 secretHash = sha256(abi.encode(secret));
+
         require(
-            status[lockId] == Status.HASH_REVEALED,
+            status[secretHash] == Status.CONFIRMED,
             "Wrong status"
         );
+
         require(
-            sha256(abi.encode(secret)) == hashlocks[lockId],
+            secretHash == swaps[secretHash].secretHash,
             "Wrong secret"
         );
 
-        status[lockId] = Status.SECRET_REVEALED;
-        secrets[lockId] = secret;
+        status[secretHash] = Status.SECRET_REVEALED;
+        secrets[secretHash] = secret;
 
-        blend.transfer(swaps[lockId].to, swaps[lockId].amount);
+        blend.transfer(swaps[secretHash].to, swaps[secretHash].amount + swaps[secretHash].fee);
     }
 
-    function claimRefund(bytes32 lockId) public {
+    function claimRefund(bytes32 secretHash) public {
         require(
-            block.timestamp >= swaps[lockId].releaseTime,
+            block.timestamp >= swaps[secretHash].releaseTime,
             "Funds still locked"
         );
-        Status st = status[lockId];
+
+        Status st = status[secretHash];
         require(
-            st == Status.INITIALIZED || st == Status.HASH_REVEALED,
+            st == Status.INITIALIZED || st == Status.CONFIRMED,
             "Wrong status"
         );
 
-        status[lockId] = Status.REFUNDED;
+        require(
+            msg.sender == swaps[secretHash].from,
+            "Sender is not the initiator"
+        );
+
+        status[secretHash] = Status.REFUNDED;
 
-        blend.transfer(swaps[lockId].from, swaps[lockId].amount);
+        blend.transfer(swaps[secretHash].to, swaps[secretHash].fee);
```

Comment:
@georgeee And why this `to` receives refund fee, haven't found explanation of this in [spec](https://www.notion.so/serokell/StakerBridge-Swap-specification-V2-4d462530e9624884b8a0f7c55720b494)

Reply:
![image](https://user-images.githubusercontent.com/25207759/98107570-c186a600-1eab-11eb-87be-bbbfc648972d.png)

Reply:
Oh, I see